alchemist-engine/src/main/java/it/unibo/alchemist/core/ArrayIndexedPriorityQueue.java

```diff
@@ -23,20 +23,18 @@
  *
  * @param <T> concentration type
  */
-public final class ArrayIndexedPriorityQueue<T> implements Scheduler<T> {
+public class ArrayIndexedPriorityQueue<T> implements Scheduler<T> {
```

Comment:
Can you retain final and implement by delegation?
Kotlin side, using `by`

Reply:
Done